examples/qml_minimal/CMakeLists.txt

```diff
@@ -78,20 +82,37 @@ add_library(${APP_NAME}_lib INTERFACE)
 # overwriting another's files.
 target_include_directories(${APP_NAME}_lib INTERFACE "${CXXQT_EXPORT_DIR}/${CRATE}")
 
-target_link_libraries(${APP_NAME}_lib INTERFACE
-    # WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
-    # otherwise the linker will discard the static variables that initialize it.
-    "$<LINK_LIBRARY:WHOLE_ARCHIVE,${CRATE}-static>"
-    Qt::Core
-    Qt::Gui
-    Qt::Qml
-    Qt::QuickControls2
-)
+if(BUILD_WASM)
+    target_link_libraries(${APP_NAME}_lib INTERFACE
+        Qt::Core
+        Qt::Gui
+        Qt::Qml
+        Qt::QuickControls2
+    )
+    
+    target_link_options(${APP_NAME}_lib INTERFACE 
+        "SHELL:-Wl,--whole-archive $<TARGET_FILE:${CRATE}-static> -Wl,--no-whole-archive"
+    )
+else()
+    target_link_libraries(${APP_NAME}_lib INTERFACE
+        # WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
+        # otherwise the linker will discard the static variables that initialize it.
+        "$<LINK_LIBRARY:WHOLE_ARCHIVE,${CRATE}-static>"
+        Qt::Core
+        Qt::Gui
+        Qt::Qml
+        Qt::QuickControls2
+    )
+endif()
 # ANCHOR_END: book_cmake_use_corrosion
 
 # ANCHOR: book_cmake_executable
 # Define the executable with the C++ source
-add_executable(${APP_NAME} cpp/main.cpp)
+if(BUILD_WASM)
+    qt_add_executable(${APP_NAME} cpp/main.cpp)
```

Comment:
assume we have to use `qt_add_executable` ? shame it was added in Qt 6 as otherwise we could just use it even for non-wasm builds. Hmmm

Reply:
Appears if you build without the `qt_add_executable` you get the `.js` and `.wasm` files but are missing the `.html` file ?

Maybe we can look into how Qt 5.x WASM was working without the newer CMake commands

Reply:
There is no Qt 5 wasm iiuc
and yea u dont get the html file, maybe there is a way around this idk